homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,249 @@
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Required(CONF_IAM_ID): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the SystemAIR platform."""
+    name = config.get(CONF_NAME)
+    device = config.get(CONF_DEVICE)
+    iam_id = config.get(CONF_IAM_ID)
+    password = config.get(CONF_PASSWORD)
+
+    climate = SystemAIRClimate(name, device, iam_id, password)
+    async_add_entities([climate])
+
+
+class SystemAIRClimate(ClimateDevice):
+    """Representation of the climate sensor."""
+
+    def __init__(self, name, unit, iam_id, password):
+        """Construct the SystemAIR Climate Device."""
+        from systemair.savecair.client import SavecairClient
+        self._name = name
+
+        """Initialize the climate device."""
+        self._support_flags = SUPPORT_TARGET_TEMPERATURE
+
+        if unit == "vtr_300":
+            self._support_flags = self._support_flags | SUPPORT_FAN_MODE
+            self._support_flags = self._support_flags | SUPPORT_OPERATION_MODE
+            self._support_flags = self._support_flags | SUPPORT_ON_OFF
+
+        # Create savecair client
+        self._client = SavecairClient(iam_id, password)
+        self._client.update_cb.append(self.update_callback)
+        self.last_operation_mode = None
+
+        self._fan_list = ['Off', 'Low', 'Normal', 'High']
+        self._operation_list = ['Auto', 'Manual', 'Crowded',
+                                'Refresh', 'Fireplace', 'Away', 'Holiday']
+
+    def update_callback(self):
+        """Update the state."""
+        self.schedule_update_ha_state()
+
+    async def async_update(self):
+        """Retrieve latest state."""
+        await self._client.update_sensors()
+        self.schedule_update_ha_state()
+
+    @property
+    def state_attributes(self):
+        """Update state attributes with additional attributes."""
+        attributes = super().state_attributes
+        attributes.update({
+            ATTR_SUPPLY_AIR_TEMPERATURE: self.supply_air_temperature,
+            ATTR_SUPPLY_AIR_FAN_SPEED: self.extract_air_fan_speed,
+            ATTR_EXTRACT_AIR_FAN_SPEED: self.supply_air_fan_speed
+        })
+        attributes.update(self._client.data)
+        return attributes
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def supply_air_fan_speed(self):
+        """Return the supply air fan speed."""
+        return self._client.data["digital_input_tacho_saf_value"] \
```

Comment:
Please rewrite all if else statements that span multiple lines to separate statements. Prefer guard clause when possible.
```py
if "digital_input_tacho_saf_value" not in self._client.data:
    return None
return self._client.data["digital_input_tacho_saf_value"]
```